urc_aruco/src/aruco_location.cpp

```diff
@@ -0,0 +1,126 @@
+#include "aruco_location.hpp"
+
+namespace aruco_location
+{
+ArucoLocation::ArucoLocation(const rclcpp::NodeOptions & options)
+: rclcpp::Node("aruco_location", options)
+{
+  location_publisher = create_publisher<urc_msgs::msg::ArucoLocation>(
+    "~/tag_location",
+    rclcpp::SystemDefaultsQoS()
+  );
+
+  //Aruco Tag Angles and Distance relative to camera.
+  aruco_subscriber = create_subscription<urc_msgs::msg::ArucoDetection>(
+    "/aruco", rclcpp::SystemDefaultsQoS(), [this](const urc_msgs::msg::ArucoDetection
+    aruco_msg) {
+      arucoCallback(aruco_msg);
+    });
+
+  gps_subscriber = create_subscription<sensor_msgs::msg::NavSatFix>(
+    "/gps/data", rclcpp::SensorDataQoS(), [this](const sensor_msgs::msg::NavSatFix gps_msg) {
+      gpsCallback(gps_msg);
+    });
+
+  //Yaw, Pitch and Roll (Orientation)
+  orientation_subscriber = create_subscription<sensor_msgs::msg::Imu>(
+    "/imu/data", rclcpp::SensorDataQoS(), [this](const sensor_msgs::msg::Imu imu_msg) {
+      orientationCallback(imu_msg);
+    });
+}
+
+
+/*
+All units need to be converted into radians
+Latitude and Longtitude calculations using Aviation Formulary V1.47:
+See --> http://www.edwilliams.org/avform147.htm#LL
+*/
+double ArucoLocation::getNextLatitude(double d, double xAngle, double yaw)
+{
+  if (!arucoRead || !gpsRead || !orientationRead) {
```

Comment:
I feel like the booleans here are exhibiting unintended behavior. `arucoRead` will never be true in the functions, and seems to be completely redundant. I also don't really understand the thinking behind `gpsRead` and `orientationRead`. Can you double check that this is all behaving the way you intended it to?

Reply:
Yeah, arucoRead was a redundant piece of code that I worked around a little lazily by just setting it true inside the if statement where the lat/lon publishing was. Definitely getting rid of it, but `gpsRead` and `orientationRead` were basically just an indicator that the gps and imu callbacks updated the cached values for their respective variables. When the arucocallback sets them false after publishing lat/lon, they just made sure that the lat/lon calculations weren't using old values and would wait for updated sensor values.

I think there is an edge case here though, where one sensor reads values when the other doesn't, keeps that cached value despite the drone moving somewhere else, then uses that old value when the broken sensor starts updating values again.